src/libs/location/src/Player.cpp

```diff
@@ -771,8 +770,9 @@ void Player::FireFromShootgun()
                 auto *const e = EntityManager::GetEntityPointer(collide->GetObjectID());
                 if (e && e != this)
                 {
-                    long n, nm;
-                    for (n = 0, nm = location->supervisor.numCharacters; n < nm; n++)
+                    long nm;
+                    size_t n;
+                    for (n = 0, nm = location->supervisor.character.size(); n < nm; n++)
```

Comment:
since you modifying the code anyway, nm is immutable, consider replacing it with "location->supervisor.character.size()" here and below:
`for (n = 0; n < location->supervisor.numCharacters; n++)`
`if (n == location->supervisor.numCharacters)`